curvefs/src/mds/fs_manager.cpp

```diff
@@ -426,22 +481,34 @@ FSStatusCode FsManager::CreateFs(const ::curvefs::mds::CreateFsRequest* request,
         LOG(ERROR) << "CreateFs fail, update fs to inited fail"
                    << ", fsName = " << fsName
                    << ", ret = " << FSStatusCode_Name(ret);
-        // TODO(wanghai): delete partiton and inode
-        // ret = metaserverClient_->DeleteInode(wrapper.GetFsId(), GetRootId());
-        // if (ret != FSStatusCode::OK) {
-        //     LOG(ERROR) << "CreateFs fail, update fs status to inited fail"
-        //                << ", then delete root inode fail"
-        //                << ", fsName = " << fsName
-        //                << ", ret = " << FSStatusCode_Name(ret);
-        //     return ret;
-        // }
-
-        // ret = fsStorage_->Delete(fsName);
-        // if (ret != FSStatusCode::OK) {
-        //     LOG(ERROR) << "CreateFs fail, insert root inode fail, "
-        //                << "then delete fs fail, fsName = " << fsName
-        //                << ", errCode = " << FSStatusCode_Name(ret);
-        // }
+        // delete partiton and inode
```

Comment:
if (request->has_recycletimehour() && request->recycletimehour() != 0) {
delete recycle dentry
delete recyle inode
}
delete root inode
delele partition
delete fs

at this order